core/txpool/legacypool/legacypool.go

```diff
@@ -565,6 +567,10 @@ func (pool *LegacyPool) validateTx(tx *types.Transaction) error {
 			if list := pool.queue[addr]; list != nil {
 				have += list.Len()
 			}
+			if pool.currentState.GetCode(addr) != nil {
```

Comment:
Maybe cheaper to check the code hash here? We have to load the account from the state to check the nonce. So loading the codehash of the account doesn't cost an extra DB access. But loading the code does, and we discard the result anyway.

Reply:
fixed

Reply:
fixed